### ISSUE TITLE
Replicas transformer add replicas to services and ingresses

### ISSUE THREAD

**Issue by benjamin-bergia** [kubernetes-sigs/kustomize#1170]
I am running kustomize master and I just noticed that using the `replicas` transformer modifies all the resources that have a matching name and a `spec` field such as Deployment, but also Ingress and Service.

When using nested overlays, the replicas count is not updated.

**Comment by benjamin-bergia** [comment]
Sorry I wasn't really clear. After adding `replicas` to my kustomization file, kustomize generate invalid resources. For example I have a deployment, a service and an ingress that have the same name `test`. After adding
```
replicas:
  - name: test
    count: 5
```
to my kustomization.yml, the `replicas` field is add to the deployment, the service and the ingress.

**Comment by damienr74** [comment]
I think this is working as intended. The replicas transformer should be able to modify any `kind:` that has the `spec: replicas:` field already. This is how it was described in #1114

I am not able to reproduce a `kind:` that does not have the `spec: replica:` field and that gets a replica added to it. Could you verify that your deployment, the service, and the ingress already have a `spec: replica:` field? Or can you provide a full set of files on which I can use to reproduce this issue?

**Comment by benjamin-bergia** [comment]
Hi @damienr74, 
here is a setup that should help you to reproduce:
```yaml
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: myapp
spec:
  replicas: 1
  template:
    spec:
      containers:
      - name: hep
        image: busybox
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: myapp
  annotations:
    kubernetes.io/ingress.class: traefik
spec:
  tls: []
  rules:
    - host: myapp.com
      http:
        paths:
          - path: /
            backend:
              serviceName: myapp
              servicePort: http
---
apiVersion: v1
kind: Service
metadata:
  name: myapp
spec:
  ports:
    - name: http
      port: 8080
      protocol: TCP
      targetPort: http
```

kustomization.yml
```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

commonLabels:
  app.kubernetes.io/name: myapp

resources:
  - deployment.yml
  - service.yml
  - ingress.yml

replicas:
  - name: myapp
    count: 5
```

When I run `kustomize build` i get:
```yaml
apiVersion: v1
kind: Service
metadata:
  labels:
    app.kubernetes.io/name: myapp
  name: myapp
spec:
  ports:
  - name: http
    port: 8080
    protocol: TCP
    targetPort: http
  replicas: 5
  selector:
    app.kubernetes.io/name: myapp
---
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app.kubernetes.io/name: myapp
  name: myapp
spec:
  replicas: 5
  selector:
    matchLabels:
      app.kubernetes.io/name: myapp
  template:
    metadata:
      labels:
        app.kubernetes.io/name: myapp
    spec:
      containers:
      - image: busybox
        name: hep
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: traefik
  labels:
    app.kubernetes.io/name: myapp
  name: myapp
spec:
  replicas: 5
  rules:
  - host: myapp.com
    http:
      paths:
      - backend:
          serviceName: myapp
          servicePort: http
        path: /
  tls: []
```

**Comment by damienr74** [comment]
Thanks for bringing this up, a fix is implemented at #1181.

/kind bug
